### PR TITLE
Multiple query tabs

### PR DESCRIPTION
**Pull request by stuzer05**
Added fully functioning query tabs

### CODE REVIEW

**Comment by stuzer05** [comment]
Minor todos:
1. add tab button needs new icon
2. add tab button needs on hover styles tweaking
3. maybe start tabs from 1 and not from 0

File: ui/main_window/central_right_widget.cpp [hunk 1]
```diff
@@ -475,10 +550,29 @@ bool CentralRightWidget::removeHostTab()
...
+                    QWidget *tab = _rootTabs->widget(i);
+                    removeTab(tab);
+                }
+            }
+        }
+
+        _queryTabTitles->clear();
+
+        _queryTabsTitleIndex = 0;
+
```
Comment (ragnar-lodbrok) [inline ui/main_window/central_right_widget.cpp:572]: Please don't do this, use BaseRootTab::tabType() to get tab's type (do pull). 
`_queryTabsTitleIndex` -- don't store logic in widgets, use models for that (models::ui::CentralRightWidgetModel)

File: ui/main_window/central_right/query/central_right_add_query_tab.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+AddQueryTab::AddQueryTab(QWidget *parent) :
+    QWidget(parent)
+{
+    createWidgets();
+}
+
+void AddQueryTab::createWidgets()
+{
+    
+}
```
Comment (ragnar-lodbrok) [inline ui/main_window/central_right/query/central_right_add_query_tab.cpp:17]: What does this empty class do?

File: ui/main_window/central_right_widget.cpp [hunk 3]
```diff
@@ -44,6 +45,8 @@ CentralRightWidget::CentralRightWidget(QWidget *parent)
                 }
             }
     );
+
+    _queryTabTitles = new QStringList();
```
Comment (ragnar-lodbrok) [inline ui/main_window/central_right_widget.cpp:49]: Why on the heap? Where is `delete`?

**Comment by ragnar-lodbrok** [comment]
Hey, thanks for your PR, but please address the issues and test it better, after several clicks I got:
image
Tabs are also too wide on first open:

File: ui/main_window/central_right_widget.cpp (hunk 1 above)
Comment (stuzer05) [reply ui/main_window/central_right_widget.cpp:572]: There are multiple problems using BaseRootTab::tabType():
1. there's no tabs list variable, tabs are stored as individual properties
2. thus the only way to foreach tabs is using _rootTabs->indexOf(i), which returns QWidget object
3. so there's no way to cast QWidget to BaseRootTab, obviously

The other problem to store tabs is that their indexes change depending on entry type user is viewing (root tabs count change, so do indexes). This is why I started to track query tabs by their title.

File: ui/main_window/central_right_widget.cpp (hunk 3 above)
Comment (stuzer05) [reply ui/main_window/central_right_widget.cpp]: Changed to stack.

File: ui/main_window/central_right_widget.cpp (hunk 1 above)
Comment (stuzer05) [reply ui/main_window/central_right_widget.cpp:572]: I guess the best way to fix both dynamic tab indexes (will remove _queryTabTitles) and workaround with tabs close button is to inherit QTabWidget.

File: ui/main_window/central_right/query/central_right_add_query_tab.cpp (hunk 2 above)
Comment (stuzer05) [reply ui/main_window/central_right/query/central_right_add_query_tab.cpp:17]: Used to distinguish between "query" tab and "add query" tab button.

**Comment by gaetan1903** [comment]
how about this PR ?

**Comment by ragnar-lodbrok** [comment]
@gaetan1903 Multiple query tabs is available in https://github.com/ragnar-lodbrok/meow-sql/releases/tag/v0.4.12-alpha

**Pull request closed without merging**